google/cloud/sql/connector/InstanceConnectionManager.py

```diff
@@ -476,7 +480,15 @@ def connect(self, driver: str, **kwargs) -> Any:
         kwargs.pop("port", None)
 
         with self._lock:
-            instance_data: InstanceMetadata = self._current.result()
+            try:
+                instance_data: InstanceMetadata = self._current.result(
+                    timeout=_instance_metadata_timeout
+                )
+            except concurrent.futures.TimeoutError:
+                raise TimeoutError(
+                    "Failed to fetch instance metadata. "
+                    f"Task timed out after {_instance_metadata_timeout}s"
+                )
 
         connect_func = {
```

Comment:
Seems like the timeout should be passed into the connector function as well?

Reply:
Interesting thing to think about: do we want the same total timeout for the whole function, or a timeout for each part (I'm guessing the first is what most people expect)

Reply:
So this might be a dumb question, but if we pass the same timeout into the connector function that we use for fetching the metadata, won't that double the timeout? Like we could wait 25 sec to get metadata and another 25 to get the connection if the timeout was 30?

Reply:
Would a better approach be to timeout for the whole function, instead of a timeout for just the call to `.result()`?